src/measure.cpp

```diff
@@ -155,6 +155,8 @@ void Measure::Reset()
     m_scoreTimeOffset.clear();
     m_realTimeOffsetMilliseconds.clear();
     m_currentTempo = MIDI_TEMPO;
+
+    m_spanningObjects.clear();
```

Comment:
These are filled in `Page::LayoutHorizontally`, aren't they? This means that they should be reset (cleared) in `Measure::ResetHorizontalAlignment`. If that does not work, then there is something not clean in the algorithm. I am thinking in particular of cases where we have a page where the `BeamSpan` is actually on the previous page. I am not absolutely sure that with the current implementation it would work because the `BeamSpan::ResolveSpanningBeamSpans` would not be called when processing the page. But maybe I am overlooking something.

Reply:
Having looked again at this it seems that this is something you would need to fill in `Doc::PrepareDrawing` because it is not expected to change when the layout is redone. That means it needs to be cleared in the counterpart or `PrepareDrawing`, namely `Measure::ResetDrawing`, which we want to happen if the content has changed (e.g., in an editor context where a note can be added or deleted.)

It seems that this is used only in `LayerElement::IsInBeamSpan`. Didn't you want to have a `bool m_isInBeamSpan` flag in `LayerElement` for that because of memory concerns?

Reply:
Those are actually filled in `Doc::PrepareDrawing`, but you're right maybe we don't even need it. I've used pointer to `beamSpan` in each `LayerElement` previously, but that was switched to this vector because of memory concerns. But having bool should be more than enough, since, as you've pointed out, it's not really used anywhere else.